src/rpc.ts

```
@@ -90,9 +178,111 @@ export class RpcMetrics {
+    // picked random int to get picked index
+    for (let i = 0; i < ranges.length; i++) {
+        const offset = ranges.slice(0, i).reduce((a, b) => a + Math.max(b, 10_000), 0);
+        const lowerBound = offset + 1;
+        const upperBound = offset + ranges[i];
+        if (lowerBound <= pick && pick <= upperBound) {
+            return i;
+        }
+    }
+
+    // out-of-range, picked value didnt match any of the items from the given list
+    return NaN;
```

Biome (1.9.4)

[error] Use Number.NaN instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.NaN instead.


(lint/style/useNumberNamespace)